par-script/src/main/java/org/ow2/parscript/PARScriptEngine.java

```diff
@@ -80,34 +81,48 @@
     private final boolean dumpErrorsIfNotForked;
 
     /**
-     * Creates or retrieves a singleton instance of the PARScriptEngine, that wraps an instance of
-     * JRIEngine.
+     * Creates or retrieves a singleton instance of the PARScriptEngine,
+     * that wraps an instance of JRIEngine.
      *
      * @return the singleton instance of the engine
      */
     public static synchronized PARScriptEngine create(PARScriptFactory factory) {
-        if (instance == null) {
-            // Check if the path to rJava is already set
-            String libPath = System.getProperty("java.library.path");
-            if (libPath == null || !libPath.contains("jri")) {
-                try {
-                    RLibPathConfigurator.configureLibraryPath();
-                } catch (Exception e) {
-                    throw new IllegalStateException("Unable to configure the library path for R", e);
-                }
-            }
-            String[] args = { "--vanilla", "--slave" };
+        if (isInForkedTask()) {
+            instance = createScriptEngine(factory);
+        } else if (instance == null) {
+            instance = createScriptEngine(factory);
```

Comment:
What is difference between instance == null and isInForkedTask ? both seem to do the same operation : instance = createScriptEngine(factory); . Maybe would be better to have a single if or to remove it completely if not needed.